node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeRepositoryProvisioner.java

```diff
@@ -113,7 +118,8 @@ public NodeRepositoryProvisioner(NodeRepository nodeRepository,
                                                                                  : requested.minResources().nodeResources();
             nodeSpec = NodeSpec.from(requested.type());
         }
-        return asSortedHosts(preparer.prepare(application, cluster, nodeSpec, groups), resources);
+        var reuseIndexes = Flags.REUSE_NODE_INDEXES.bindTo(flagSource).with(FetchVector.Dimension.ZONE_ID, zone.systemLocalValue()).value();
```

Comment:
Not sure if it's intended, but if you do it like this the flag value will only be read once, when this is constructed.

To read it every time you should pass the `BooleanFlag` to `prepare` and call `.value()` there.

Reply:
The line is in the prepare method.

Reply:
My mistake. 🙂